snowfall/training/ctc_graph.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c)  2020  Xiaomi Corp.       (author: Fangjun Kuang)
+
+from functools import lru_cache
+from typing import (
+    Iterable,
+    List,
+)
+
+import torch
+import k2
+import _k2
```

Comment:
I’m not sure but importing _k2 outside of k2 code base seems like an antipattern, maybe something needs to be included in public k2 API?

Reply:
I agree.

If we can move `compose` to k2, then we do not need `import _k2`.

Reply:
Fixed.